plugin/trino-databend/src/main/java/io/trino/plugin/databend/DatabendClient.java

```diff
@@ -0,0 +1,758 @@
+    }
+
+    @Override
+    public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, List<JdbcSortItem> sortOrder)
+    {
+        for (JdbcSortItem sortItem : sortOrder) {
+            Type sortItemType = sortItem.column().getColumnType();
+            checkArgument(!(sortItemType instanceof CharType), "Unexpected char type: %s", sortItem.column().getColumnName());
+            if (sortItemType instanceof VarcharType) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    protected Optional<TopNFunction> topNFunction()
+    {
+        return Optional.of((query, sortItems, limit) -> {
+            String orderBy = sortItems.stream().map(sortItem -> {
+                String ordering = sortItem.sortOrder().isAscending() ? "ASC" : "DESC";
+                String nullsHandling = sortItem.sortOrder().isNullsFirst() ? "NULLS FIRST" : "NULLS LAST";
+                return format("%s %s %s", quoted(sortItem.column().getColumnName()), ordering, nullsHandling);
+            }).collect(joining(", "));
+            return format("%s ORDER BY %s LIMIT %d", query, orderBy, limit);
+        });
+    }
+
+    @Override
+    public boolean isTopNGuaranteed(ConnectorSession session)
+    {
+        return true;
+    }
+
+    @Override
+    public Optional<PreparedQuery> implementJoin(
+            ConnectorSession session,
+            JoinType joinType,
+            PreparedQuery leftSource,
+            Map<JdbcColumnHandle, String> leftProjections,
+            PreparedQuery rightSource,
+            Map<JdbcColumnHandle, String> rightProjections,
+            List<ParameterizedExpression> joinConditions,
+            JoinStatistics statistics)
+    {
+        if (joinType == JoinType.FULL_OUTER) {
+            return Optional.empty();
+        }
+        return implementJoinCostAware(
+                session,
+                joinType,
+                leftSource,
+                rightSource,
+                statistics,
+                () -> super.implementJoin(session, joinType, leftSource, leftProjections, rightSource, rightProjections, joinConditions, statistics));
+    }
+
+    @Override
+    public ResultSet getTables(Connection connection, Optional<String> schemaName, Optional<String> tableName)
+            throws SQLException
+    {
+        DatabaseMetaData metadata = connection.getMetaData();
+        if (tableName.isPresent()) {
+            // Databend maps their "database" to SQL catalogs and does not have schemas
+            return metadata.getTables(schemaName.orElse(null), null, tableName.get(), getTableTypes().map(types -> types.toArray(String[]::new)).orElse(null));
+        }
+        else {
+            return metadata.getTables(schemaName.orElse(null), null, null, getTableTypes().map(types -> types.toArray(String[]::new)).orElse(null));
+        }
+    }
+
+    @Override
+    protected String getTableSchemaName(ResultSet resultSet)
```

Comment:
Remove a redundant override. This logic is the same as its super method.

Reply:
Fixed.

Reply:
This is not fixed.

Reply:
Sorry , some comments were hidden so I miss some. Fixed now.